Source/DafnyLanguageServer/Plugins/DafnyCodeAction.cs

```diff
@@ -0,0 +1,52 @@
+﻿using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.Dafny.LanguageServer.Plugins;
+
+/// <summary>
+/// Describe a single suggestion made to the user, where edits can be
+/// computed lazily.
+/// If the edits can be provided instantly, you can use the derived class
+/// `new InstantDafnyCodeAction(title, edits)`
+/// </summary>
+public abstract class DafnyCodeAction {
```

Comment:
I would like to avoid this, since I know some implementers of code actions in F# and since it's an abstract class, they would need to implement boilerplate methods such as clone, copy, etc.
But even if you make it concrete, I'm not sure how the Func<>  interfaces which are C# specific would translate to F#. So, let's avoid the complexity and keep the API as simple as possible.

Reply:
I think it might be good to take some time to agree on what language we're writing the API for. You seem to have settled on a particular style of API that sits between what is idiomatic for C# and F#, while I'm still trying to optimise for C#.

In this particular case, wouldn't `record DafnyCodeAction(string Title, Container<Diagnostic> Diagnostics, Func<DafnyCodeActionEdit[]> GetEdits)` work well for F# as well, since there is no abstract class, and F# can use a lambda expression for the `GetEdits` argument?